numpy/lib/tests/test_function_base.py

```diff
@@ -2688,6 +2692,63 @@ def test_with_minlength_and_weights(self):
         w = np.array([0.2, 0.3, 0.5, 0.1, 0.2])
         y = np.bincount(x, w, 8)
         assert_array_equal(y, np.array([0, 0.2, 0.5, 0, 0.5, 0.1, 0, 0]))
+    
+    def test_with_out(self):
+        x = np.array([1, 5, 2, 4, 1])
+        out = np.ones(6, dtype=int)
```

Comment:
I'd maybe fill with `-1` to make sure the test shows that bins that are never used are indeed untouched (your test does it too, but this would make it more obvious).